metricflow/model/objects/elements/identifier.py

```diff
@@ -42,30 +43,25 @@ class Identifier(HashableBaseModel):
     identifiers: List[CompositeSubIdentifier] = []
     expr: Optional[str] = None
 
-    def __init__(  # type: ignore
-        self,
-        name: str,
-        type: IdentifierType,
-        role: Optional[str] = None,
-        entity: Optional[str] = None,
-        identifiers: Optional[List[CompositeSubIdentifier]] = None,
-        expr: Optional[str] = None,
-        **kwargs: Dict[str, Any],  # the parser may instantiate objects with additional fields (eg __parsing_context__)
-    ) -> None:
-        """Normal pydantic initializer except we set entity to name"""
-
-        identifiers = identifiers or []
-
-        super().__init__(
-            name=name,
-            type=type,
-            role=role,
-            entity=entity,
-            identifiers=identifiers,
-            expr=expr,
-        )
-        if self.entity is None:
-            self.entity = self.reference.element_name
+    @validator("entity", always=True)
+    @classmethod
+    def default_entity_value(cls, value: Any, values: Any) -> str:  # type: ignore[misc]
+        """Defaulting the value of the identifier 'entity' value using pydantic validator
+
+        If an entity value is provided that is a string, that will become the value of
+        entity. If the provifed entity value is None, the entity value becomes the
+        element_name representation of the identifier's name.
+        """
+
+        if value is None:
+            if "name" not in values:
+                raise ValueError("Failed to default entity value because objects name value was not defined")
+            return IdentifierReference(element_name=values["name"]).element_name
```

Comment:
Do we need to do this reference construction? We basically drop it on the floor after this statement, and there's no real validation benefit to the IdentifierReference initializer since Pydantic will validate the "name" field on its own.

Reply:
I didn't know if there was any magic that meant sometimes doing `IndentifierReference(...).element_name` would ever not be the name. It looks like currently that is the case. We just need to be aware of if the initialization of `IndentifierReference` ever in the future begins using any fancy logic to modify the `element_name`